cmake/EthOptions.cmake

```diff
@@ -11,14 +11,59 @@ macro(configure_project)
 	option(VMTRACE "Enable VM tracing" OFF)
 	option(PROFILING "Enable profiling (deprecated)" OFF)
 	option(FATDB "Enable fat state database" ON)
-	option(ROCKSDB "Build with RocksDB instead of LevelDB" OFF)
 	option(PARANOID "Enable additional checks when validating transactions (deprecated)" OFF)
 	option(MINIUPNPC "Build with UPnP support" OFF)
 	option(FASTCTEST "Enable fast ctest" OFF)
 
+	set(supported_dbs LEVELDB ROCKSDB LMDB)
+	string(REPLACE ";" ", " supported_dbs_str "${supported_dbs}")
+	set(DB LEVELDB CACHE STRING "Database implementation to use (${supported_dbs_str})")
+	set_property(CACHE DB PROPERTY STRINGS ${supported_dbs})
+	set(valid_db OFF)
+	foreach(supported_db ${supported_dbs})
```

Comment:
Don't do this validation. If CMake does not support it out of the box, forget it.